swap/swap.go

```diff
@@ -92,7 +363,118 @@ func (s *Swap) loadState(peer *protocols.Peer) (err error) {
 	return
 }
 
-//Clean up Swap
-func (swap *Swap) Close() {
-	swap.stateStore.Close()
+//loadCheque loads the last cheque for a peer from the state store (persisted)
+func (s *Swap) loadCheque(peer enode.ID) (err error) {
+	//only load if the current instance doesn't already have this peer's
+	//last cheque in memory
+	var cheque *Cheque
+	if _, ok := s.cheques[peer]; !ok {
+		err = s.stateStore.Get(peer.String()+"_cheques", &cheque)
+		s.cheques[peer] = cheque
+	}
+	return
+}
+
+// saveLastReceivedCheque loads the last received cheque for peer
+func (s *Swap) loadLastReceivedCheque(peer enode.ID) (cheque *Cheque) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	s.stateStore.Get(peer.String()+"_cheques_received", &cheque)
+	return
+}
+
+// saveLastReceivedCheque saves cheque as the last received cheque for peer
+func (s *Swap) saveLastReceivedCheque(peer enode.ID, cheque *Cheque) error {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.stateStore.Put(peer.String()+"_cheques_received", cheque)
+}
+
+// Close cleans up swap
+func (s *Swap) Close() {
+	s.stateStore.Close()
```

Comment:
not sure if you are supposed to call this if you're using the normal state store that the rest of swarm is using. if you are - then the store should be closed on the application level on shutdown and this can be omitted

Reply:
? 

This is not  global state store but a `StateStore` with a dedicated file for balances.
As part of the RPC server, it should be closed on a clean shutdown, and I think this is the best place.

Maybe it should not  be called `s.stateStore` but `balanceStore` or something

Reply:
Renamed to just "store"